src/benchmarks/micro/libraries/System.Text.Json/Utf8JsonReader/Perf.Get.cs

```diff
@@ -0,0 +1,183 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Text.Json.Tests
+{
+    [BenchmarkCategory(Categories.Libraries, Categories.JSON)]
+    public class Perf_Get
+    {
+        private byte[] _jsonSByteBytes;
+        private byte[] _jsonInt16Bytes;
+        private byte[] _jsonInt32Bytes;
+        private byte[] _jsonInt64Bytes;
+
+        private byte[] _jsonSingleBytes;
+        private byte[] _jsonDoubleBytes;
+        private byte[] _jsonDecimalBytes;
+
+        private byte[] _jsonStringBytes;
+        private byte[] _jsonGuidBytes;
+        private byte[] _jsonDateTimeBytes;
+        private byte[] _jsonDateTimeOffsetBytes;
+
+        [Params(true, false)]
+        public bool UseLargeValue;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            if (UseLargeValue)
+            {
+                _jsonSByteBytes = Encoding.UTF8.GetBytes(sbyte.MaxValue.ToString());
+                _jsonInt16Bytes = Encoding.UTF8.GetBytes(short.MaxValue.ToString());
+                _jsonInt32Bytes = Encoding.UTF8.GetBytes(int.MaxValue.ToString());
+                _jsonInt64Bytes = Encoding.UTF8.GetBytes(long.MaxValue.ToString());
+
+                _jsonSingleBytes = Encoding.UTF8.GetBytes(float.MaxValue.ToString("F"));
+                _jsonDoubleBytes = Encoding.UTF8.GetBytes(double.MaxValue.ToString("F"));
+                _jsonDecimalBytes = Encoding.UTF8.GetBytes(decimal.MaxValue.ToString("F"));
+                _jsonStringBytes = Encoding.UTF8.GetBytes($"\"{new string('a', 1024 * 2)}\"");
+            }
+            else
+            {
+                _jsonSByteBytes = Encoding.UTF8.GetBytes(default(sbyte).ToString());
+                _jsonInt16Bytes = Encoding.UTF8.GetBytes(default(short).ToString());
+                _jsonInt32Bytes = Encoding.UTF8.GetBytes(default(int).ToString());
+                _jsonInt64Bytes = Encoding.UTF8.GetBytes(default(long).ToString());
+
+                _jsonSingleBytes = Encoding.UTF8.GetBytes(default(float).ToString());
+                _jsonDoubleBytes = Encoding.UTF8.GetBytes(default(double).ToString());
+                _jsonDecimalBytes = Encoding.UTF8.GetBytes(default(decimal).ToString());
+                _jsonStringBytes = Encoding.UTF8.GetBytes("\"The quick brown fox jumps over the lazy dog.\"");
+            }
+
+            _jsonGuidBytes = Encoding.UTF8.GetBytes($"\"{Guid.Empty}\"");
+            _jsonDateTimeBytes = Encoding.UTF8.GetBytes($"\"{DateTime.MaxValue:O}\"");
+            _jsonDateTimeOffsetBytes = Encoding.UTF8.GetBytes($"\"{DateTimeOffset.MaxValue:O}\"");
+        }
+
+        [Benchmark]
+        public byte GetByte()
+        {
+            var reader = new Utf8JsonReader(_jsonSByteBytes);
+            reader.Read();
+            return reader.GetByte();
+        }
```

Comment:
@Jozkee do you know what part of the total time is spent for the creation of `Utf8JsonReader`, calling `Read` and calling `GetByte()`?  What is the typical usage of `reader.GetX()` methods? 

I am asking because the creation of the reader and call to `reader.Read` might be more expensive than the call to `GetX` which we are trying to measure here. If this is true, we should reconsider rewriting the benchmark to something like:

```cs
public sbyte ReadAndGetSBytes()
{
    var reader = new Utf8JsonReader(_jsonSByteBytes); // _jsonSByteBytes should contain more than 1 SByte (100 should be enough)
    sbyte result = default;
    while (reader.Read())
    {
        result += reader.GetSByte();
    }
    return result;
}
```

this should allow for removal of `UseLargeValue` and hence reducing the number of new benchmarks and the time it takes to run them by half. Please see https://github.com/dotnet/performance/blob/master/docs/microbenchmark-design-guidelines.md#Test-Cases for more

Reply:
Good point, for Int32, ~80% time is spent in the first two lines:
```cs
[Benchmark]
public void CreateAndReadInt32()
{
    var reader = new Utf8JsonReader(_jsonInt32Bytes);
    reader.Read();
}
```

|             Method | UseLargeValue |     Mean |    Error |   StdDev |   Median |      Min |      Max | Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------- |-------------- |---------:|---------:|---------:|---------:|---------:|---------:|------:|------:|------:|----------:|
| **CreateAndReadInt32** |         **False** | **39.95 ns** | **0.548 ns** | **0.512 ns** | **39.81 ns** | **39.42 ns** | **41.11 ns** |     **-** |     **-** |     **-** |         **-** |
|           GetInt32 |         False | 46.41 ns | 0.302 ns | 0.283 ns | 46.47 ns | 45.90 ns | 46.81 ns |     - |     - |     - |         - |
| **CreateAndReadInt32** |          **True** | **49.73 ns** | **0.211 ns** | **0.187 ns** | **49.71 ns** | **49.48 ns** | **50.02 ns** |     **-** |     **-** |     **-** |         **-** |
|           GetInt32 |          True | 62.31 ns | 0.647 ns | 0.573 ns | 62.42 ns | 60.74 ns | 63.15 ns |     - |     - |     - |         - |

Reply:
Good point, for `GetInt32`, ~80% time is spent in the first two lines:
```cs
[Benchmark]
public void CreateAndReadInt32()
{
    var reader = new Utf8JsonReader(_jsonInt32Bytes);
    reader.Read();
}
```

|             Method | UseLargeValue |     Mean |    Error |   StdDev |   Median |      Min |      Max | Gen 0 | Gen 1 | Gen 2 | Allocated |
|------------------- |-------------- |---------:|---------:|---------:|---------:|---------:|---------:|------:|------:|------:|----------:|
| **CreateAndReadInt32** |         **False** | **39.95 ns** | **0.548 ns** | **0.512 ns** | **39.81 ns** | **39.42 ns** | **41.11 ns** |     **-** |     **-** |     **-** |         **-** |
|           GetInt32 |         False | 46.41 ns | 0.302 ns | 0.283 ns | 46.47 ns | 45.90 ns | 46.81 ns |     - |     - |     - |         - |
| **CreateAndReadInt32** |          **True** | **49.73 ns** | **0.211 ns** | **0.187 ns** | **49.71 ns** | **49.48 ns** | **50.02 ns** |     **-** |     **-** |     **-** |         **-** |
|           GetInt32 |          True | 62.31 ns | 0.647 ns | 0.573 ns | 62.42 ns | 60.74 ns | 63.15 ns |     - |     - |     - |         - |

Reply:
FWIW I added `ReadToEnd*` benchmarks in order to measure how much time is spent in traversing the 100 elements of the json.

Reply:
Another approach that we could take to avoid having the `ReadtoEnd*` methods could be just call `reader.GetInt3()2` over the same token a hundred times, that could remove the overhead of `reader.Read()`.